### PR TITLE
feat: cherry-pick audit findings from closed PR #31

### PR DESCRIPTION
**Pull request by Questi0nM4rk**
## Summary

Cherry-picks the 11 surviving changes from closed PR #31 (fix/audit-findings), cleanly rebased onto current main after PRs #33 and #34 landed.

### Changes

- **Portable shebangs**: `#!/bin/bash` → `#!/usr/bin/env bash` for all 8 hook scripts (NixOS/Guix compatibility)
- **uv-first installer**: Prefer `uv tool install` over `pipx`/`pip` for pre-commit, ruff, mypy, bandit, vulture, pip-audit
- **Init improvements**:
  - Backup files before overwriting on `--force` (creates `.bak`)
  - `--dry-run` flag to preview what `ai-guardrails init` would do
  - Dual PreToolUse hooks: `protect-generated-configs` + `dangerous-command-check`
  - Generic `_install_pretooluse_hook()` helper (DRY)
- **Remove review-all workflow** (superseded by `ai-guardrails comments --resolve-all`)
- **pyproject.toml fixes**:
  - Add `ai-guardrails = "guardrails.cli:main"` entry point
  - Fix wheel build config (`packages` + `sources` mapping)
- **Pre-commit version bumps**: gitleaks v8.30.0, semgrep v1.151.0, bandit 1.9.3, ruff v0.15.1, mypy v1.19.1, shellcheck v0.11.0.1
- **CI template**: `pip install` → `uv tool install`, generic coverage/pyright paths
- **New**: `.clang-tidy` config for C/C++ projects
- **New tests**: `test_init_integration.py` (10 tests) + `test_template_sync.py` (1 test)

### What was dropped from PR #31

- `coderabbit.py` changes (file deleted in PR #33, replaced by `comments.py`)
- Duplicate hook/CLI work already merged in PRs #33/#34

## Test plan

- [x] `uv run pytest tests/ -v` — 429 tests pass
- [x] `uv run ruff check lib/python/guardrails/` — clean
- [x] `uv run pyright` — 0 errors
- [x] `pre-commit run --all-files` — all pass
- [ ] Review bot feedback addressed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added ai-guardrails CLI entry and --dry-run for init
  * Included additional pre-commit/pre-push hooks and dangerous-command checks
  * Added a clang-tidy configuration

* **Improvements**
  * Prefer a modern installer tool with fallbacks and clearer error messaging
  * Hook scripts made more portable (env-based interpreter)
  * Create backups when overwriting configs
  * Updated packaging sources and wheel declaration
  * Bumped several pre-commit hook versions

* **Tests**
  * Added init integration and template sync tests

* **Documentation**
  * Added enterprise assessment and status-command plan documents

### CODE REVIEW

File: lib/python/guardrails/init.py [hunk 1]
```diff
@@ -568,6 +589,79 @@ def _append(filepath: Path) -> None:
         _print_ok("Created CLAUDE.md with guardrails rules")
 
 
+# ---------------------------------------------------------------------------
+# Dry-run report
+# ---------------------------------------------------------------------------
+
+
+def _dry_run_report(
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply lib/python/guardrails/init.py]: Acknowledged — _would() is intentionally scoped to _dry_run_report().

File: lib/python/guardrails/init.py [hunk 2]
```diff
@@ -292,47 +301,73 @@ def _configure_pip_audit(mode: str, project_dir: Path) -> None:
...
     pre_tool_use.append(hook_entry)
 
-    # Write settings (backup first if file exists)
     settings_path.parent.mkdir(parents=True, exist_ok=True)
     if settings_path.exists():
         backup_path = settings_path.with_suffix(".json.bak")
         shutil.copy2(settings_path, backup_path)
     settings_path.write_text(json.dumps(settings, indent=2) + "\n")
-    _print_ok("Claude Code PreToolUse hook installed")
+    _print_ok(f"{label} installed")
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply lib/python/guardrails/init.py]: This is a CLI tool run by a single user, not a concurrent service. Atomic writes add complexity without benefit here.

File: tests/test_init_integration.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+from unittest.mock import patch
+
+import pytest
+from guardrails.init import run_init
+
+_PATCHES = (
+    "guardrails.init._install_precommit_hooks",
+    "guardrails.init._install_claude_hook",
+    "guardrails.init._install_dangerous_cmd_hook",
+)
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply tests/test_init_integration.py]: Fixed in ea8e982 — extracted _mock_hooks fixture with @pytest.mark.usefixtures.

File: lib/python/guardrails/init.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply lib/python/guardrails/init.py]: Thank you.

File: tests/test_init_integration.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    assert rc == 0
+    # No files should be created
+    assert not (project_dir / ".pre-commit-config.yaml").exists()
+    assert not (project_dir / ".editorconfig").exists()
+    assert not (project_dir / "ruff.toml").exists()
+    # But the report should list planned actions
+    captured = capsys.readouterr()
+    assert "would" in captured.out
+    assert ".editorconfig" in captured.out
+    assert "ruff.toml" in captured.out
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply tests/test_init_integration.py]: Addressed: edge-case tests added where practical.

File: tests/test_template_sync.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    return None
+
+
+def test_claude_review_prompt_sync() -> None:
+    """Verify .github/ and templates/ Claude review prompts match."""
+    github_wf_path = ROOT / ".github" / "workflows" / "claude-code-review.yml"
+    template_wf_path = ROOT / "templates" / "workflows" / "claude-review.yml"
+
+    if not github_wf_path.exists() or not template_wf_path.exists():
+        pytest.skip("Workflow files not found")
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply tests/test_template_sync.py]: Fixed: _extract_prompt now iterates all jobs.

File: tests/test_template_sync.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    github_wf = yaml.safe_load(github_wf_path.read_text())
+    template_wf = yaml.safe_load(template_wf_path.read_text())
+
+    github_prompt = _extract_prompt(github_wf)
+    template_prompt = _extract_prompt(template_wf)
```
[the remark this answers is not in the document]
Comment (Questi0nM4rk) [reply tests/test_template_sync.py]: Fixed: added isinstance guard before _extract_prompt calls.
